library/std/src/io/buffered/bufwriter.rs

```diff
@@ -241,6 +242,26 @@ impl<W: Write> BufWriter<W> {
         &self.buf
     }
 
+    /// Returns
+    ///
+    /// * the number of currently buffered bytes
+    /// * the whole buffer allocation, i.e. buffered data plus unused capacity
+    /// * the inner writer
+    ///
+    /// The amount of currently buffered data tracked by `BufWriter` itself is set to zero and
+    /// will not be flushed, thus the caller must ensure all currently buffered data gets written
+    /// to the inner writer.
+    pub(in crate::io) fn as_raw_parts(&mut self) -> (usize, &mut [MaybeUninit<u8>], &mut W) {
```

Comment:
This should be unsafe as the caller could de-initialize already buffered data.

Reply:
It sets the length to zero, thus the data is already assumed to be not initialized after the borrows have been used.

Reply:
It seems pretty unexpected that a method called "as_raw_parts" would modify the raw parts.